cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/bulkwriter/RecordWriter.java

```diff
@@ -110,20 +133,42 @@ public StreamResult write(Iterator<Tuple2<DecoratedKey, Object[]>> sourceIterato
         Map<String, Object> valueMap = new HashMap<>();
         try
         {
+            Set<Range<BigInteger>> newRanges = initialTokenRangeMapping.getRangeMap().asMapOfRanges().entrySet()
+                                                                       .stream()
+                                                                       .map(Map.Entry::getKey)
+                                                                       .collect(Collectors.toSet());
```

Comment:
nit: this is simpler to get a copy of the key set

```suggestion
            Set<Range<BigInteger>> newRanges = new HashSet<>(initialTokenRangeMapping.getRangeMap().asMapOfRanges().keySet());
```

Reply:
makes sense.